scripts/modals/popups/speedometer-select.js

```diff
@@ -3,20 +3,49 @@ class SpeedometerSelectPopup {
 	static container = $('#SpeedometerSelectContainer');
 	/** @static @type {TextEntry} */
 	static textEntry = $('#SpeedometerName');
+	/** @type {Label} @static */
+	static invalidNameLabel = $('#InvalidNameLabel');
 	static selected = 0;
+	static speedometerNames = [];
+
+	static onTextSubmitted() {
+		const text = this.textEntry.text;
+		if (text === '' || text.includes(',')) {
+			SpeedometerSelectPopup.invalidNameSubmitted();
+			return;
+		}
+
+		let passed = true;
+		SpeedometerSelectPopup.speedometerNames.every((speedometerName) => {
+			if (text.toUpperCase() === speedometerName.toUpperCase()) {
+				passed = false;
+				return false;
+			}
+			return true;
+		});
+		if (!passed) {
+			SpeedometerSelectPopup.invalidNameSubmitted();
+			return;
+		}
 
-	static onAddButtonPressed() {
 		const callbackHandle = $.GetContextPanel().GetAttributeInt('callback', -1);
-		if (callbackHandle !== -1)
-			UiToolkitAPI.InvokeJSCallback(
-				callbackHandle,
-				SpeedometerSelectPopup.selected,
-				SpeedometerSelectPopup.textEntry.text
-			);
+		if (callbackHandle !== -1) UiToolkitAPI.InvokeJSCallback(callbackHandle, SpeedometerSelectPopup.selected, text);
 		UiToolkitAPI.CloseAllVisiblePopups();
 	}
 
+	static invalidNameSubmitted() {
+		SpeedometerSelectPopup.invalidNameLabel.visible = true;
```

Comment:
Is anything setting this back to invisible?

Reply:
Nothing besides just closing and reopening the popup, should there be a way to reset the invalid name popup? I can do it after a certain amount of time or I was thinking there could just be an animation to indicate that there was another invalid attempt at adding a speedometer.

Reply:
Oh I see. There could probably be some event listener for when the text entry becomes empty or something in the future, but this is only if people get annoyed by it not going away when correcting their input, so don't worry about it for now. The `init()` setting it to false works fine.